### ISSUE TITLE
[Accessibility]: Batches: Batch change label isn't descriptive

### ISSUE THREAD

**Issue by BolajiOlajide** [sourcegraph/sourcegraph#34669]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34069

### Problem description

When navigating through the list of batch changes, the label indicating the status of the batch change isn't descriptive.
It currently reads `DRAFT` / `OPEN` e.t.c

<img width="1147" alt="Screenshot 2022-04-28 at 21 30 33" src="https://user-images.githubusercontent.com/25608335/165840411-f7950772-4cf4-4259-b109-8e26f161ce00.png">

### Expected behavior

The status text should read `Draft Batch Change` or whatever status the batch change is in.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
